src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -242,6 +245,34 @@ public boolean reconnect()
 		return true;
 	}
 
+	/**
+	 * Simulates a Player consuming an Edible Item
+	 * @param consumable The Item to consume
+	 */
+	public void simulateConsumeItem(@NotNull ItemStack consumable)
```

Comment:
This doesn't completely act the same way Paper does. Please look at `net.minecraft.world.entity.LivingEntity#completeUsingItem()` for Paper's implementation.

Reply:
What do you mean

Reply:
Sorry, I wrote that a bit hastily.

First some sounds are played at `net.minecraft.world.entity.LivingEntity#completeUsingItem` line 3843

A GenericGameEvent should be fired secondly. `net.minecraft.world.entity.LivingEntity#startUsingItem(InteractionHand, boolean)` line 3766, and `net.minecraft.server.level.ServerLevel#gameEvent` line 1570.

And lastly if the PlayerItemConsumeEvent is cancelled, another GenericGameEvent should be fired. `net.minecraft.world.entity.LivingEntity#completeUsingItem` line 3854, `#stopUsingItem` line 2920, and `net.minecraft.server.level.ServerLevel#gameEvent` line 1570.